builders/sbt.go

```diff
@@ -147,7 +150,38 @@ func (builder *SBTBuilder) IsModule(target string) (bool, error) {
 	return false, errors.New("IsModule is not implemented for SBTBuilder")
 }
 
-// InferModule is not implemented
-func (builder *SBTBuilder) InferModule(target string) (module.Module, error) {
-	return module.Module{}, errors.New("InferModule is not implemented for SBTBuilder")
+// DiscoverModules returns a root build.sbt if found, and build configs for all sub-projects otherwise
+func (builder *SBTBuilder) DiscoverModules(dir string) ([]config.ModuleConfig, error) {
+	_, err := os.Stat(filepath.Join(dir, "build.sbt"))
+	if err == nil {
+		// in a multi-project build (https://www.scala-sbt.org/1.x-beta/docs/Multi-Project.html) we return the root build.sbt only
+		absDir, err := filepath.Abs(dir)
+		if err != nil {
+			absDir = dir
+		}
+		artifactName := filepath.Base(absDir)
+		return []config.ModuleConfig{
+			config.ModuleConfig{
+				Name: artifactName,
+				Path: "build.sbt",
+				Type: "sbt",
+			},
+		}, nil
+	}
+
+	// no pom in root directory; find and parse all of them
+	sbtFilePaths, err := doublestar.Glob(filepath.Join(dir, "**", "build.sbt"))
+	if err != nil {
+		return nil, err
+	}
+	moduleConfigs := make([]config.ModuleConfig, len(sbtFilePaths))
+	for i, path := range sbtFilePaths {
+		artifactName := filepath.Dir(path) // use the dirname as it's too hard to parse from build.sbt
```

Comment:
Isn't `build.sbt` a Scala file? That makes it literally impossible to parse without executing.

Reply:
No, you can grep with reasonable success rate (maybe 75%?) like we do with the ruby implementation; from what i've seen in scala code so far.

Reply:
I meant impossible in general. It's a bit misleading for this comment to suggest it's just "too hard" (unless you would consider the halting problem to be the same level of "too hard").